### ISSUE TITLE
`flee` and `launch` in conversations in the `on complete` clause

### ISSUE THREAD

**Issue by jjhankins** [endless-sky/endless-sky#3167]
It would be nice sometimes to be able to launch immediately on completion of a mission (e.g., an `infiltrating` mission where there's nothing left to do but launch), but neither `flee` nor `launch` have any effect in the `on complete` clause of a mission.

**Comment by endless-sky** [comment]
If the mission isn't the very end of a story thread, you can just put the conversation into the next mission in the thread and have that one launch you.

**Comment by jjhankins** [comment]
Yes, and for threads where it is the final mission I have a "now that you're done you launch" mission, which should take care of the problem with other threads starting as well since they'd be marked `minor`. So this is more of a wishlist item.

**Comment by Amazinite** [comment]
Instead of giving the last mission an `on complete`, create a new mission and move the `on complete` conversation of the old last mission to the `to offer` of the new last mission and put `flee` at the end so that the mission declines.

**Comment by jjhankins** [comment]
Then you're giving up on substitutions like the payment amount and cargo.

**Comment by jjhankins** [comment]
They're random, as I'm using them. I've got a series of deliveries to isolated communities that are in rebellion against the pirate warlords.

**Comment by Amazinite** [comment]
```
on accept
	event "trigger flee"
[...]
on fail
	clear "event: trigger flee"
on complete
	<payment>
	[conversation || dialog]

mission "flee from pirates
	landing
	source <destination of the previous mission>
	to offer
		has "event: trigger flee"
	on offer
		clear "event: trigger flee"
		conversation
			`Get out.`
				flee
```
This might work for what you want; a separate conversation should pop up that triggers the flee. You just can't have this conversation have any text substitutions pertinent to the previous mission.

**Comment by jjhankins** [comment]
That's pretty much what I have, only the condition is set in the `on complete` of the main mission - it still works and there's no need to clear it `on fail`. Then all my missions that have that structure trigger that follow-up. It works, the only annoyance is the extra screen of text.

**Comment by endless-sky** [comment]
Fixed by merging #3420.
